src/util/settings.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/settings.h>
+
+#include <univalue.h>
+
+namespace util {
+namespace {
+
+struct Source {
+    SettingsSpan span;
+    bool forced = false;
+    bool config_file = false;
+    bool config_file_default_section = false;
+
+    explicit Source(SettingsSpan span) : span(span) {}
+    Source& SetForced() noexcept { forced = true; return *this; }
+    Source& SetConfigFile(bool default_section) noexcept { config_file = true; config_file_default_section = default_section; return *this; }
+};
+
+template <typename Fn>
+static void MergeSettings(const Settings& settings, const std::string& section, const std::string& name, Fn&& fn)
+{
+    if (auto* value = FindKey(settings.forced_settings, name)) {
+        fn(Source(SettingsSpan(*value)).SetForced());
+    }
+    if (auto* values = FindKey(settings.command_line_options, name)) {
+        fn(Source(SettingsSpan(*values)));
+    }
+    if (!section.empty()) {
+        if (auto* map = FindKey(settings.ro_config, section)) {
+            if (auto* values = FindKey(*map, name)) {
+                fn(Source(SettingsSpan(*values)).SetConfigFile(/* default_section= */ false));
+            }
+        }
+    }
+    if (auto* map = FindKey(settings.ro_config, "")) {
+        if (auto* values = FindKey(*map, name)) {
+            fn(Source(SettingsSpan(*values)).SetConfigFile(/* default_section= */ true));
+        }
+    }
+}
+} // namespace
+
+SettingsValue GetSetting(const Settings& settings,
+    const std::string& section,
+    const std::string& name,
+    bool ignore_default_section_config,
+    bool get_chain_name)
+{
+    SettingsValue result;
+    MergeSettings(settings, section, name, [&](Source source) {
+        // Weird behavior preserved for backwards compatibility: Apply negated
+        // setting in otherwise ignored sections. A negated value in the
```

Comment:
So the behavior you're describing is in case of negated value in default section, this result is going to take precedence on value in network section even it's a non-negated one ? If yes saying that network specific options are overriden would be more correct than ignored.

Reply:
Thanks, after your explanation, I think comment is good enough (or don't suggestion in mind to make it better)